src/motor_hardware.cc

```diff
@@ -142,6 +144,27 @@ void MotorHardware::readInputs() {
                     }
                     break;
                 }
+                case MotorMessage::REG_BATTERY_VOLTAGE: {
+                    int32_t data = mm.getData();
+                    sensor_msgs::BatteryState bstate;
+                    bstate.voltage = (float)data * pid_params.battery_voltage_multiplier +
+                                   pid_params.battery_voltage_offset;
+                    bstate.current = std::numeric_limits<float>::quiet_NaN();
+                    bstate.charge = std::numeric_limits<float>::quiet_NaN();
+                    bstate.capacity = std::numeric_limits<float>::quiet_NaN();
+                    bstate.design_capacity = std::numeric_limits<float>::quiet_NaN();
+                    if (bstate.voltage < 22.0) { // 18V = 0%,  22V = 12%
+                        bstate.percentage = std::max(0.0, 2.0 * (bstate.voltage - 18.0));
+                    }
+                    else { // 22V = 12%, 24V = 42%, 26V = 72%. 27V = 87%
+                        bstate.percentage = std::min(100.0, 12.0 + (bstate.voltage - 22.0) * 15.0);
```

Comment:
Same issue, should be something like `std::min(1.0, 0.12 + (bstate.voltage - 22.0) * 0.15);`